### DISCUSSION TITLE
Mail Merge

### DISCUSSION THREAD

**Discussion by andres-gimenez** [typst/typst#4079] [Q&A]
How do I generate multiple documents from a template and data source? The data source can be a tom or json file, no database is required.

Is it possible to pass a parameter when generating a document that indicates a file as the data source and a file name to generate?

**Comment by istudyatuni** [comment]
See sys.inputs: https://typst.app/docs/reference/foundations/sys

**Reply by andres-gimenez** [reply]
I suppose that to create several documents you will have to create a script.

**Reply by OD-tpeko** [reply]
Yes. Or you could create a function with your document content and invoke it multiple times based on some `.json` content you can parse from within Typst itself..
